cdap-watchdog/src/main/java/io/cdap/cdap/logging/ErrorLogsClassifier.java

```diff
@@ -184,7 +219,8 @@ private Map<String, String> getChildTags(ErrorClassificationResponse response) {
       errorCategory = errorCategory.substring(0, errorCategory.length() - stageName.length() - 1);
     }
     ImmutableMap.Builder<String, String> tags = ImmutableMap.builder();
-    tags.put(Metrics.Tag.ERROR_CATEGORY, errorCategory);
+    tags.put(Metrics.Tag.ERROR_CATEGORY, Strings.isNullOrEmpty(response.getRuleId()) ? errorCategory
+        : String.format("%s-'%s'", errorCategory, response.getRuleId()));
```

Comment:
Why do you need to quote it? is it going to be machine-readable? If so, I'd better sanitize it

Reply:
Also it look a bit strange that we are removing stage name and replacing it with rule id. What is the goal here? It's better not to reuse same tag/format for different purposes as it becomes very confusing during analysis

Reply:
BTW: I also see you use quoting in populate response, so your parsing here won't work with quoted value

Reply:
`ruleId` will be absent, if `stageName` is present in the response because we prioritize response from `failureDetailsProvider` over `rule-based` classification.

Reply:
we are removing `stageName` only from metrics to reduce cardinality here, as it is not much informative for metrics, it will have `errorCategory` as `plugin` which is enough to analyse the errors from `plugin` .

Reply:
As discussed offline, updated.